From 8b461b409df81711e0b67d909271e7a254fe1a69 Mon Sep 17 00:00:00 2001
From: Youff <asa.jazal@gmail.com>
Date: Fri, 20 Mar 2026 14:09:50 +0700
Subject: [PATCH] Update ksu.c

---
 KernelSU/kernel/ksu.c | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/KernelSU/kernel/ksu.c b/KernelSU/kernel/ksu.c
index 7c49214..affdef3 100644
--- a/KernelSU/kernel/ksu.c
+++ b/KernelSU/kernel/ksu.c
@@ -3,7 +3,6 @@
 #include <linux/kobject.h>
 #include <linux/module.h>
 #include <linux/workqueue.h>
-#include <linux/version.h>
 
 #include "allowlist.h"
 #include "app_profile.h"
@@ -147,10 +146,10 @@ module_exit(kernelsu_exit);
 MODULE_LICENSE("GPL");
 MODULE_AUTHOR("weishu");
 MODULE_DESCRIPTION("Android KernelSU");
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 0, 0)
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 13, 0)
-MODULE_IMPORT_NS("VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver");
-#else
-MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
-#endif
+#ifdef MODULE_IMPORT_NS
+    #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 13, 0)
+        MODULE_IMPORT_NS("VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver");
+    #else
+        MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
+    #endif
 #endif
